### DISCUSSION TITLE
How can I change the time to ask random questions?

### DISCUSSION THREAD

**Discussion by jonnytri53** [ardha27/AI-Waifu-Vtuber#63] [Q&A] [answered]
Hi, community. Every 3-4 minutes, the chatbot will randomly make up a question and answer it automatically. But I can't find out where it is. Can anyone help me? Thank you so much.

**Comment by ardha27** [comment]
As long as i remember, i didnt put that kind of logic inside my code

**Reply by jonnytri53** [reply]
It's kind of odd. I didn't say anything for entirely 30 minutes, and it appeared some random text. Anyway, by any chance I can make the chatbot ask me a question every 2 minutes? Question 2: Can I remove ctrl-shift to ask questions, but it will be default recording like we have the conversations between two real humans? Thank you, dev.

My conversation.json
```
{
            "role": "user",
            "content": "Jonny said Thank you for watching. I'm your host, Joe Rogan. We'll see you next week."
        },
        {
            "role": "assistant",
            "content": "Thanks, Joe! Have a great week! ;)"

 {
            "role": "user",
            "content": "Jonny said "
        },
        {
            "role": "assistant",
            "content": "Hey Jonny, is there anything specific you wanted to talk about? ;)"
        },
        {
            "role": "user",
            "content": "Jonny said "
        },
        {
            "role": "assistant",
            "content": "Hey Jonny, what's on your mind? ;)"
        },
        {
            "role": "user",
            "content": "Jonny said "
        },
        {
            "role": "assistant",
            "content": "Is there anything specific you'd like to chat about, Jonny? ;)"
```

**Reply by ardha27** [reply, marked as answer]
I didnt add add error handling on that side. You can change the code so it only process message that contain text and not processing empty message

1. Yes you can. Maybe you can add time flag when the last coversation occured, if it already pass 2 mnt. Make a code that will send instruction to openai (on background) to generate a random topic to talk about

2. Yes, but you need to change record audio function. You can find library that can use microphone threshold so it only recording when there are someone speaking

**Comment by ardha27** [comment]
I think its occur because the conversation is to long, and the chatbot/openai got confused on how to respond that long conversation
